pkg/cloud/service.go

```diff
@@ -95,16 +95,13 @@ func (s *Service) pollOperation(ctx context.Context, op operation) error {
 		pollCount++
 		klog.V(5).Infof("op.isDone(%v) waiting; op = %v, poll count = %d", ctx, op, pollCount)
 		s.RateLimiter.Accept(ctx, op.rateLimitKey())
-		done, err := op.isDone(ctx)
-		if err != nil {
-			klog.V(5).Infof("op.isDone(%v) error; op = %v, poll count = %d, err = %v, retrying", ctx, op, pollCount, err)
-		}
-
-		if done {
-			break
+		switch done, err := op.isDone(ctx); {
+		case err != nil:
+      klog.V(5).Infof("op.isDone(%v) error; op = %v, poll count = %d, err = %v, retrying", ctx, op, pollCount, err)
+			return err
+		case done:
+      klog.V(5).Infof("op.isDone(%v) complete; op = %v, poll count = %d, op.err = %v", ctx, op, pollCount, op.error())
+			return op.error()
```

Comment:
missing return value when !done?

Reply:
What do you mean? This function is intended to poll-loop, in particular the previous version continued on !done

Reply:
If !done, then it loops again which is expected right?

Reply:
ok -- my diff was not in unified so I mismatched the braces